libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP.h

```diff
@@ -121,12 +137,15 @@ typedef enum ePING_REPLY_STATUS
 	eInvalidData		/* A reply was received to an outgoing ping but the payload of the reply was not correct. */
 } ePingReplyStatus_t;
 
-typedef enum eNETWORK_ADDRESS_TYPE 
+typedef struct xIP_TIMER
 {
-	eNetWorkAddressTypeIPV4,
-	eNetWorkAddressTypeIPV6,
-	eNetWorkAddressTypeHostName
-} eNetWorkAddressType_t;
+	uint32_t
+		bActive : 1,	/* This timer is running and must be processed. */
+		bExpired : 1;	/* Timer has expired and a task must be processed. */
+	TimeOut_t xTimeOut;
```

Comment:
`TimeOut_t` not defined in this scope?

Reply:
In general, it is assumed that the FreeRTOS header files are included before the FreeRTOS+TCP header files.
`TimeOut_t` is defined in `task.h`.
Can you tell in which module you see this compilation error?

Reply:
I think this might be in espressif project only: I am getting the following error when compiling the source for espressif,

```
../vendors/espressif/boards/esp32/components/freertos_tcpip/tcpip_adapter/tcpip_adapter.c
In file included from ../vendors/espressif/boards/esp32/components/freertos_tcpip/tcpip_adapter/include/tcpip_adapter.h:44:0,
                 from ../vendors/espressif/boards/esp32/components/freertos_tcpip/tcpip_adapter/tcpip_adapter.c:18:
../libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP.h:146:2: error: unknown type name 'TimeOut_t'
```
This is I guess an error in CMakeFile which includes the +TCP/include before it includes the FreeRTOS header files.

To make my file compile, I had to include `task.h` in `FreeRTOS_IP.h`. (I was just trying to get it to compile. This solution is in no way a good solution.)

I'll try to pinpoint the issue.